index.bs

```diff
@@ -294,79 +325,76 @@ This callback will be invoked when there are changes to <i>target</i>'s intersec
+            true.</li>
+        <li>Post a task to <a>notify intersection observers</a>, or
+            enqueue a task to <a>notify intersection observers</a> in the
+            <a>list of idle request callbacks</a> with an appropriate
             <i>timeout</i>.
 
             Issue: Should we define an appropriate <i>timeout</i>?
         </li>
     </ol>
 
-<h4 id='3-2-2-invoke-intersection-observer-callback'>
-<dfn>Invoke an IntersectionObserverCallback</dfn>
-</h4>
-    To <a>invoke an IntersectionObserverCallback</a> for <i>observer</i>, run
-    these steps:
+<h4 id='3-2-2-notify-intersection-observers'>Notify Intersection Observers</h4>
+    To <dfn>notify intersection observers</dfn>, run these steps:
     <ol>
-        <li>If <i>observer</i>'s internal {{[[QueuedEntries]]}} slot is empty,
-            return.</li>
-        <li>Construct a <i>sequence</i>&lt;{{IntersectionObserverEntry}}> from
-            {{[[QueuedEntries]]}}.</li>
-        <li>Clear <i>observer</i>'s internal {{[[QueuedEntries]]}} slot.</li>
-        <li>Remove <i>observer</i> from <i>Document</i>'s internal
-            {{[[PendingIntersectionObserverCallbacks]]}} slot.</li>
-        <li>Invoke <i>callback</i> with the
-            <i>sequence</i>&lt;{{IntersectionObserverEntry}}> as
-            the first argument and <i>observer</i> as the second argument.</li>
+        <li>Set the internal <a>\[[IntersectionObserverTaskQueued]]</a> slot to
+            false.</li>
+        <li>Let <i>notify list</i> be a copy of
+            <a>unit of related similar-origin browsing contexts</a>'s internal
+            <a>\[[IntersectionObservers]]</a> slot.</li>
+        <li>For each {{IntersectionObserver}} object <i>observer</i> in
+            <i>notify list</i>, run these steps:</li>
+            <ol>
+                <li>If <i>observer</i>'s internal {{[[QueuedEntries]]}} slot is
+                    empty, continue.</li>
+                <li>Let <i>queue</i> be a copy of <i>observer</i>'s internal
+                    {{[[QueuedEntries]]}} slot.
+                </li>
+                <li>Clear <i>observer</i>'s internal {{[[QueuedEntries]]}} slot.
+                </li>
+                <li>Invoke <i>callback</i> with <i>queue</i> as the first
+                argument and <i>observer</i> as the second argument and
+                <a>callback this value</a>. If this throws an exception,
+                <a>report the exception</a>.</li>
+            </ol>
     </ol>
 
-<h4 id='3-2-3-queue-intersection-observer-entry'>
-<dfn>Queue an IntersectionObserverEntry</dfn>
-</h4>
-    To <a>queue an IntersectionObserverEntry</a> for <i>observer</i>, run
+<h4 id='3-2-3-queue-intersection-observer-entry'>Queue an
+IntersectionObserverEntry</dfn></h4>
+    To <dfn>queue an IntersectionObserverEntry</dfn> for <i>observer</i>, run
     these steps:
     <ol>
         <li>Construct an {{IntersectionObserverEntry}}.</li>
```

Comment:
Is this enough?
Construct an *IntersectionObserverEntry*, passing in *time, rootBounds, boundingClientRect, intersectionRect* and *target*.